src/main/kotlin/io/libp2p/pubsub/gossip/GossipScore.kt

```diff
@@ -0,0 +1,213 @@
+package io.libp2p.pubsub.gossip
+
+import io.libp2p.core.PeerId
+import io.libp2p.core.pubsub.ValidationResult
+import io.libp2p.etc.types.cappedDouble
+import io.libp2p.etc.types.createLRUMap
+import io.libp2p.etc.types.millis
+import io.libp2p.etc.util.P2PService
+import pubsub.pb.Rpc
+import java.util.Optional
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.ScheduledFuture
+import java.util.concurrent.TimeUnit
+import kotlin.math.max
+import kotlin.math.min
+import kotlin.math.pow
+
+class GossipScore(
+    val params: GossipScoreParams = GossipScoreParams(),
+    val executor: ScheduledExecutorService,
+    val curTime: () -> Long
```

Comment:
nit:
```suggestion
    val curTimeInMillis: () -> Long
```

Reply:
If you don't mind I would change to `curTimeMillis`